src/plugins/vis_type_vega/public/data_model/vega_parser.ts

```diff
@@ -90,12 +135,12 @@ export class VegaParser {
     this.tooltips = this._parseTooltips();
 
     this._setDefaultColors();
-    this._parseControlPlacement(this._config);
+    this._parseControlPlacement();
     if (this.useMap) {
       this.mapConfig = this._parseMapConfig();
-    } else if (this.spec.autosize === undefined) {
+    } else if (this.spec?.autosize === undefined) {
```

Comment:
Is this `this.spec?` and the `this.spec!` below necessary? 
The 
```
if (!this.spec) {
      this.spec = hjson.parse(this.rawSpec, { legacyRoot: false });
    }
```

should make sure it's always set, right?

Reply:
Yes, but as the spec can be undefined in that case `?` and `!` is necessary. Should I replace those with `this.spec && this.spec.autosize === undefined` 
I have a suggestion here why could we just set the value of this.spec right inside the constructor?. so we could avoid the chance of `undefined` from everywhere